### DISCUSSION TITLE
The go2rtc service exited with code 1 (by signal 0) [go2rtc v1.5.0]

### DISCUSSION THREAD

**Discussion by Sch0field9** [blakeblackshear/frigate#6949] [Ask A Question] [answered]
Hi there,
sorry I don't know if im correct here or should go to go2rtc. But i need some help, and i saw someone had a similar problem before, just not the same.

go2rtc starts, but then crashes instantly. ofc go2rtc dashboard is not reachable and all streams fail

I followed the steps in the frigate docs to implement go2rtc v1.5 but i get the following logs:
```
 [WARN] Using go2rtc binary from '/config/go2rtc' instead of the embedded one
 [INFO] Starting go2rtc...
 ./run.user: line 70: /config/go2rtc: Success
 [INFO] The go2rtc service exited with code 1 (by signal 0)
``` 

any help would be much apreciated

---------------------------------------------------------------------------
INFOS:
this was what i did:
```shell
- cd /config <-- only when i tried it directly in the container
- curl -O https://github.com/AlexxIT/go2rtc/releases/download/v1.5.0/go2rtc_linux_amd64
- mv go2rtc_linux_amd64 go2rtc
- chmod +x go2rtc
``` 
Running in proxmox CT debian amd65:

my docker compose:
```dockerfile
version: "3.9"
services:
  frigate:
    container_name: frigate
    privileged: true 
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:stable
    cap_add:
      - CAP_PERFMON
    shm_size: "2100mb"
    devices:
      - /dev/dri/renderD128
      - /dev/dri/
    device_cgroup_rules:
      - 'c 189:* rmw' 
      - 'c 188:* rmw'
    volumes:
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
      - /root/frigate/config.yml:/config/config.yml
      - /root/frigate/go2rtc:/config/go2rtc
      - /var/lib/docker/volumes/frigate_recordings/_data:/media/frigate
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "5000:5000"

      - "8554:8554"
    environment:
      LIBVA_DRIVER_NAME: i965
      FRIGATE_RTSP_PASSWORD: "************"
      PLUS_API_KEY: 42134546jg5f32ghjkl4k5j43hj2kl325k46j5j3k42546jn57m3l2k3lk2j46bh5
```

**Comment by NickM-27** [comment]
This usually happens when execute permission was not set for the file correctly

**Comment by Sch0field9** [comment]
thank you for your suggestion.
I went throu all the permissions i could think of
chmod +xrw
chmod o+xrw /config/go2rtc

etc.. and every combination of it

ls -ld now shows full permission on the directory and the file itself:
drwxrwxrwx 

however i still get the same error...

**Reply by NickM-27** [reply]
try chmod 777 go2rtc

**Reply by Sch0field9** [reply]
actually didn't try that one, only 705...

but nope still same error

**Reply by NickM-27** [reply]
what device are you running this on?

**Reply by Sch0field9** [reply]
proxmox CT debian amd64
Host is on intel Raptor Lake (13th gen)

**Reply by Sch0field9** [reply]
my docker dir looks like this /root/frigate/ which contains config.yml and go2rtc
in my container dir /config/ i can also see config.yml and go2rtc
i executed all command in my docker console and in my container console....

**Comment by Sch0field9** [comment, marked as answer]
ok I'm a bone head... but I figured it out...
when i curled the build i forgot the -L. And because of the redirect it didn't download anything. It didnt give me an error message, but created the file anyway. I didnt check if something was actually downloaded.
so i tried with an empty file the whole time.

TL:DR Solution:

```shell
curl -O -L https://github.com/AlexxIT/go2rtc/releases/download/v1.5.0/go2rtc_linux_amd64
``` 
fixed the problem
